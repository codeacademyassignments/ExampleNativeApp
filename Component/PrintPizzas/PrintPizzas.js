import React from 'react';
import { Text, TextInput, View } from 'react-native';
import PrintPizzasStyle from './PrintPizzas.style';

export default class PrintPizzas extends React.Component {
  state = {
    text: '',
  }

  translateToPizzas() {
    return this.state.text.split(' ').map(() => '🍕').join(' ');
  }

  render() {
    return (
      <View style={{ flex: 1, alignItems: 'center' }}>
        <TextInput style={PrintPizzasStyle.textToTranslate} placeholder="Enter text to translate" onChangeText={text => this.setState({ text })} />
        <Text style={PrintPizzasStyle.translatedText}>{this.translateToPizzas()}</Text>
      </View>
    );
  }
}
