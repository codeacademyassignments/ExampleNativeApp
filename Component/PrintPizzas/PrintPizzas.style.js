import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  textToTranslate: { height: 100, fontSize: 25 },
  translatedText: { fontSize: 42 },
});
