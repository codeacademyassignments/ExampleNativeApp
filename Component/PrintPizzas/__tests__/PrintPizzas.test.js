import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import PrintPizzas from '../PrintPizzas';
import '../../../setupTests';

describe('PrintPizzas', () => {
  it('should match old snapshot', () => {
    const tree = renderer.create(<PrintPizzas />);
    expect(tree).toMatchSnapshot();
  });
  it('should change state on text change in input', () => {
    const wrapper = shallow(<PrintPizzas />);
    expect(wrapper.instance().state.text).toEqual('');
    wrapper.find('TextInput').simulate('changeText', 'WOw!');
    expect(wrapper.instance().state.text).toEqual('WOw!');
  });
});

describe('translateToPizzas', () => {
  it('should generate pizzas as number of space into state', () => {
    const wrapper = shallow(<PrintPizzas />);
    expect(wrapper.instance().state.text).toEqual('');
    wrapper.find('TextInput').simulate('changeText', 'WOw!');
    expect(wrapper.instance().state.text).toEqual('WOw!');
    expect(wrapper.instance().translateToPizzas()).toEqual('üçï');
  });
});
