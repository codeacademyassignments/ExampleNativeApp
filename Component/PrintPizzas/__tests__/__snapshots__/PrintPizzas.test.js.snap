// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintPizzas should match old snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter text to translate"
    style={
      Object {
        "fontSize": 25,
        "height": 100,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "fontSize": 42,
      }
    }
  >
    üçï
  </Text>
</View>
`;
